app/middleware/postcodeValidator.js

```diff
@@ -1,48 +1,42 @@
 const log = require('../lib/logger');
 const postcodeValidator = require('../lib/postcodeValidator');
-const isNotEnglishPostcode = require('../lib/isNotEnglishPostcode');
+const renderer = require('./renderer');
 const PostcodesIOClient = require('postcodesio-client');
 
 const PostcodesIO = new PostcodesIOClient();
 
 function handleError(error, postcode, res, next) {
-  postcodeValidator.handlePostcodeError(error, postcode, res, next);
+  renderer.postcodeError(error, postcode, res, next);
 }
 
-function validatePostcode(result, postcode, req, res, next) {
+function render(result, postcode, req, res, next) {
   if (!result) {
-    postcodeValidator.renderInvalidPostcodePage(postcode, req, res);
+    renderer.invalidPostcodePage(postcode, req, res);
   } else {
     next();
   }
 }
 
-function validateEnglishLocation(req, res, next) {
+function validatePostcode(req, res, next) {
   const postcode = res.locals.postcode;
 
   if (postcodeValidator.isOutcode(postcode)) {
     log.info('validate-outcode-skip');
     next();
   } else {
-    log.info('validate-location-start');
+    log.info('validate-postcode-start');
     PostcodesIO
       .validate(postcode)
-      .then(result => validatePostcode(result, postcode, req, res, next))
+      .then(result => render(result, postcode, req, res, next))
       .catch(error => handleError(error, postcode, res, next));
-    log.info('validate-location-end');
+    log.info('validate-postcode-end');
   }
 }
 
-function validateLocation(req, res, next) {
+module.exports = (req, res, next) => {
   if (res.locals.postcode) {
-    if (isNotEnglishPostcode(res.locals.postcode)) {
-      postcodeValidator.renderPostcodeNotEnglish(res.locals.postcode, req, res);
```

Comment:
One issue I have with this is that for a full out of England postcode (that starts with the regex) we are doing to calls to the API now, while before we were filtering it out at start. 

If the overhead is acceptable for less code, than it's fine by me.